core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -76,7 +82,29 @@
     public RuntimeClassLoader(ClassLoader parent, List<Path> applicationClassesDirectories, Path frameworkClassesDirectory,
             Path transformerCache) {
         super(parent);
-        this.applicationClassDirectories = applicationClassesDirectories;
+        List<Path> nonEmptyAppClasses = new ArrayList<>();
```

Comment:
Any chance you could call this one `nonEmptyApplicationClassDirectories` to be consistent?

It makes it clear it's the path of the directories, not the ones of the classes. Otherwise, it can be pretty confusing.